app/queries/display_ads/filtered_ads_query.rb

```diff
@@ -18,22 +18,31 @@ def initialize(area:, user_signed_in:, organization_id: nil, article_tags: [],
       @article_tags = article_tags
       @article_id = article_id
       @permit_adjacent_sponsors = permit_adjacent_sponsors
+      @user_tags = user_tags
     end
 
     def call
       @filtered_display_ads = approved_and_published_ads
       @filtered_display_ads = placement_area_ads
 
-      if @article_tags.any?
-        @filtered_display_ads = tagged_post_comment_ads
+      if @article_id.present?
+        if @article_tags.any?
+          @filtered_display_ads = tagged_ads(@article_tags)
+        end
+
+        if @article_tags.blank?
+          @filtered_display_ads = untagged_ads
+        end
+
+        @filtered_display_ads = unexcluded_article_ads
       end
 
-      if @article_tags.blank?
-        @filtered_display_ads = untagged_post_comment_ads
+      if @user_tags.present? && @user_tags.any?
+        @filtered_display_ads = tagged_ads(@user_tags)
       end
 
-      if @article_id.present?
-        @filtered_display_ads = unexcluded_article_ads
+      if @user_tags.blank? && DisplayAd::HOME_FEED_PLACEMENTS.include?(@area)
```

Comment:
I think we could move the logic into a helper method to keep a more consistent level of abstraction. I have an attempt of that below, however, I'm a bit unclear on the logic around `untagged_ads`, it seems like it might ought to be an `||` not an `&&` ?

```suggestion
      if @user_tags.blank? && placement_area_allows_untagged?
        @filtered_display_ads = untagged_ads
      elsif @user_tags.present? && @user_tags.any?
        @filtered_display_ads = tagged_ads(@user_tags)
      end
```

Reply:
> however, I'm a bit unclear on the logic around untagged_ads, it seems like it might ought to be an || not an && ?

Done ✅ 

We use an && because if user tags are blank **and** we are rendering a home feed placement then we should filter by untagged tags. 

We _do not want_ to filter by untagged tags if only the user tags are blank for any placement area like for example the sidebar ad. 

I think this goes back to only implementing these conditions for the placement area in the feed for now. 

Does this make sense?